### PR TITLE
Resolve unused variables in tests

### PR DESCRIPTION
**Pull request by kianasun**
Variables such as `device` and `sparse` in for loops should be used in tests.

### CODE REVIEW

**Review by ezyang — CHANGES_REQUESTED** [review]
Tests failures are real

```

Aug 09 07:29:23 ERROR: test_bool_tensor_comparison_ops (__main__.TestTorch)
Aug 09 07:29:23 ----------------------------------------------------------------------
Aug 09 07:29:23 Traceback (most recent call last):
Aug 09 07:29:23   File "test_torch.py", line 3487, in test_bool_tensor_comparison_ops
Aug 09 07:29:23     self.assertEqual(a == torch.tensor(True, dtype=torch.bool), torch.tensor([1, 0, 1, 0, 1, 0], dtype=torch.uint8))
Aug 09 07:29:23 RuntimeError: Expected object of backend CUDA but got backend CPU for argument #2 'other'
```

File: test/test_torch.py [hunk 1]
```diff
@@ -3474,23 +3474,25 @@ def test_tensor_factory_cuda_type(self):
...
             self.assertEqual(a != b, torch.tensor([0, 0, 0, 1, 0, 1], dtype=torch.uint8))
             self.assertEqual(a < b, torch.tensor([0, 0, 0, 1, 0, 1], dtype=torch.uint8))
             self.assertEqual(a > b, torch.tensor([0, 0, 0, 0, 0, 0], dtype=torch.uint8))
             self.assertEqual(a >= b, torch.tensor([1, 1, 1, 0, 1, 0], dtype=torch.uint8))
             self.assertEqual(a <= b, torch.tensor([1, 1, 1, 1, 1, 1], dtype=torch.uint8))
             self.assertEqual(a > False, torch.tensor([1, 0, 1, 0, 1, 0], dtype=torch.uint8))
-            self.assertEqual(a == torch.tensor(True, dtype=torch.bool), torch.tensor([1, 0, 1, 0, 1, 0], dtype=torch.uint8))
-            self.assertEqual(a == torch.tensor(0, dtype=torch.bool), torch.tensor([0, 1, 0, 1, 0, 1], dtype=torch.uint8))
+            self.assertEqual(a == torch.tensor(True, dtype=torch.bool, device=device),
+                             torch.tensor([1, 0, 1, 0, 1, 0], dtype=torch.uint8))
```
Comment (xuhdev) [inline test/test_torch.py]: The second parameter should set `device` (and the type can also be changed to `bool`).

Comment (kianasun) [reply test/test_torch.py]: @xuhdev Do you mean second parameters of all `assertEqual`s in this function?

Comment (xuhdev) [reply test/test_torch.py]: Yes

**Review by xuhdev — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**